extensions/podman/src/extension.ts

```diff
@@ -253,8 +276,78 @@ function execPromise(command: string, args?: string[], logger?: extensionApi.Log
   });
 }
 
+async function getLastRunInfo(storagePath: string): Promise<PodmanInfo | undefined> {
+  const podmanInfoPath = path.resolve(storagePath, 'podman-ext.json');
+  if (!fs.existsSync(storagePath)) {
+    await promisify(fs.mkdir)(storagePath);
+  }
+
+  if (!fs.existsSync(podmanInfoPath)) {
+    return undefined;
+  }
+
+  try {
+    const infoBuffer = await readFile(podmanInfoPath);
+    return JSON.parse(infoBuffer.toString('utf8'));
+  } catch (err) {
+    console.error(err);
+  }
+
+  return undefined;
+}
+
+interface InstalledPodman {
+  version: string;
+}
+
+export async function getPodmanInstallation(): Promise<InstalledPodman | undefined> {
+  try {
+    const versionOut = await execPromise(getPodman(), ['--version']);
+    const versionArr = versionOut.split(' ');
+    const version = versionArr[versionArr.length - 1];
+    return { version };
+  } catch (err) {
+    // no podman binary
+    return undefined;
+  }
+}
+
 export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
   storedExtensionContext = extensionContext;
+  const podmanInfoRaw = await getLastRunInfo(extensionContext.storagePath);
+  podmanInfo = new PodmanInfoImpl(podmanInfoRaw, extensionContext.storagePath);
+  const bundledPodmanVersion = getBundledPodmanVersion();
+
+  if (!podmanInfoRaw) {
+    const installedPodman = await getPodmanInstallation();
+    if (!installedPodman) {
```

Comment:
probably this whole stuff should be in a dedicated file as it's related to a different part